### ISSUE TITLE
`Naming/VariableNumber` mistakes `:x86_64` for a "symbol number"

### ISSUE THREAD

**Issue by postmodern** [rubocop/rubocop#11604]
I noticed that the `Naming/VariableNumber` rule was mistaking a symbol `:x86_64` which represents the x86-64 architecture, as a symbol number. I believe this is incorrect, as the symbol begins with an `x`, which makes it an alpha-numeric symbol.

## Expected behavior

Ignore `:x86_64` symbols.

## Actual behavior

```
$ bundle exec rubocop --debug                          
For /home/postmodern/test/ruby/rubocop: configuration from /home/postmodern/test/ruby/rubocop/.rubocop.yml                                                       
Default configuration from /home/postmodern/test/ruby/rubocop/vendor/bundle/ruby/3.1.0/gems/rubocop-1.45.1/config/default.yml                                    
Use parallel by default.                                              
Running parallel inspection
Loading cache from /home/postmodern/.cache/rubocop_cache/da11a5156d303827c8892095e0ccf80789af8f4c/6d7a3b621ca1730e04accd938619e4bdab66cfb1/41c3597a28be60f36846d434087d6cc466b1f00f
Loading cache from /home/postmodern/.cache/rubocop_cache/da11a5156d303827c8892095e0ccf80789af8f4c/6d7a3b621ca1730e04accd938619e4bdab66cfb1/16b1d2cd5a129139444f764f3f107677640b95f9
Inspecting 2 files
Scanning /home/postmodern/test/ruby/rubocop/Gemfile
Loading cache from /home/postmodern/.cache/rubocop_cache/da11a5156d303827c8892095e0ccf80789af8f4c/6d7a3b621ca1730e04accd938619e4bdab66cfb1/16b1d2cd5a129139444f764f3f107677640b95f9
CScanning /home/postmodern/test/ruby/rubocop/test.rb
Loading cache from /home/postmodern/.cache/rubocop_cache/da11a5156d303827c8892095e0ccf80789af8f4c/6d7a3b621ca1730e04accd938619e4bdab66cfb1/41c3597a28be60f36846d434087d6cc466b1f00f
W

Offenses:

Gemfile:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
source 'https://rubygems.org'
^
test.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
arch = :x86_64
^
test.rb:1:8: C: Naming/VariableNumber: Use normalcase for symbol numbers.
arch = :x86_64
       ^^^^^^^
test.rb:3:1: W: Lint/Debugger: Remove debugger entry point p arch.
p arch
^^^^^^

2 files inspected, 4 offenses detected, 2 offenses autocorrectable
Finished in 0.3252425979990221 seconds
```

## Steps to reproduce the problem

`Gemfile:`

```ruby
source 'https://rubygems.org'

gem 'rubocop'
```

`test.rb`:

```ruby
arch = :x86_64

p arch
```

1. `bundle install`
2. `bundle exec rubocop --debug`

## RuboCop version

```
$ bundle exec rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.26.0, running on ruby 3.1.3) [x86_64-linux]
```
